server/service/client.go

```diff
@@ -468,6 +468,25 @@ func (c *Client) ApplyGroup(
 	}
 
 	if specs.AppConfig != nil {
+		// Clear out SSO settings if not provided.
+		if _, ok := specs.AppConfig.(map[string]interface{})["sso_settings"]; !ok {
```

Comment:
(cautious) nit: This would panic in the case `specs.AppConfig` is not `map[string]interface{}`.

You should first cast `specs.AppConfig` to `map[string]interface{}` and then attempt to access the key.
```go
appConfig, ok := specs.AppConfig.(map[string]interface{})
if !ok {
  return nil, nil, nil, nil, fmt.Errorf("unexpected app config type: %T", specs.AppConfig)
}
if _, ok := appConfig["sso_settings"]; !ok {
  // ...
}

// etc.
```

Reply:
Good catch!  I wrapped it in a `if _, ok := specs.AppConfig.(map[string]interface{}); ok {` because [other similar code](https://github.com/fleetdm/fleet/blob/49289d13c9fd2a7c88853e9a3da646b72d3159f7/server/service/client.go#L1261-L1264) returns without error in this case.  I'm not sure if that should ever be expected but just going with the pattern for now.